Mage.Sets/src/mage/cards/i/InfectiousRage.java

```diff
@@ -0,0 +1,116 @@
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{R}");
+        this.subtype.add(SubType.AURA);
+
+        // Enchant creature
+        TargetPermanent auraTarget = new TargetCreaturePermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.BoostCreature));
+        Ability ability = new EnchantAbility(auraTarget);
+        this.addAbility(ability);
+
+        // Enchanted creature gets +2/-1.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new BoostEnchantedEffect(2, -1)));
+        
+        // When enchanted creature dies, choose a creature at random Infectious Rage can enchant.
+        // Return Infectious Rage to the battlefield attached to that creature.
+        this.addAbility(new DiesAttachedTriggeredAbility(new InfectiousRageReattachEffect(), "enchanted creature"));
+        
+    }
+
+    private InfectiousRage(final InfectiousRage card) {
+        super(card);
+    }
+
+    @Override
+    public InfectiousRage copy() {
+        return new InfectiousRage(this);
+    }
+}
+
+class InfectiousRageReattachEffect extends OneShotEffect {
+
+    public InfectiousRageReattachEffect() {
+        super(Outcome.PutCardInPlay);
+        this.staticText = "choose a creature at random {this} can enchant. Return {this} to the battlefield attached to that creature.";
+    }
+
+    public InfectiousRageReattachEffect(final InfectiousRageReattachEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public InfectiousRageReattachEffect copy() {
+        return new InfectiousRageReattachEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+
+        Player controller = game.getPlayer(source.getControllerId());
+        Card auraCard = game.getCard(source.getSourceId());
+        Permanent auraPermanent = (Permanent) game.getLastKnownInformation(source.getSourceId(), Zone.BATTLEFIELD);
+        if (controller == null || auraCard == null || auraPermanent == null) {
+            return false;
+        }
+
+        FilterPermanent filter = new FilterPermanent();
+        filter.add(CardType.CREATURE.getPredicate());
+        filter.add(new PermanentCanBeAttachedToPredicate(auraPermanent)); // Doesn't yet exclude creatures with protection abilities
+        List<Permanent> permanents = game.getBattlefield().getActivePermanents(filter, source.getControllerId(), source, game);
+
+        if (!permanents.isEmpty()) {
+            Permanent creature = permanents.get(RandomUtil.nextInt(permanents.size()));
+            if (creature != null) {
+                StringBuilder message = new StringBuilder(creature.getLogName());
+                message.append(" was randomly chosen from among: ");
+                for (Permanent p : permanents) {
+                    message.append(p.getLogName());
+                    message.append(", ");
+                }
```

Comment:
you should use a stream and `Collectors.joining(", ")` here to generate the message, also I don't think the message is really necessary anyway

Reply:
Yeah, I'll just remove it